libcst/matchers/_visitors.py

```diff
@@ -366,6 +372,8 @@ def _visit_matchers(
     node: cst.CSTNode,
     metadata_resolver: cst.MetadataDependent,
 ) -> Dict[BaseMatcherNode, Optional[cst.CSTNode]]:
+    # Populate matchers dict if it's empty
+    matchers = matchers or _gather_matchers(metadata_resolver)
```

Comment:
Looks like we'll pay extra on every `on_visit()` now if there are no matchers declared in the visitor/transformer. Would be worth eliminating that